### PR TITLE
FlutterDriver: deprecate enableAccessibility; redirect it to setSemantics; add setSemantics tests

### LINKED ISSUE #65192: FlutterDriver: Support enabling accessibility on Mobile web (iOS-Safari)
## Use case
After https://github.com/flutter/flutter/pull/65051, developers will be able to enable accessibility on Desktop web. However, the way it enables accessibility may NOT work on Mobile web. In fact, it works on Android-chrome but not iOS-Safari.

## Proposal

Find a way to enable accessibility on iOS-Safari.

### PR DESCRIPTION
**Pull request by yjbanov**
On mobile `FlutterDriver.enableAccessibility` never worked. On the web it has the same effect as `setSemantics`, except that it used to rely on an implementation detail that's not cross-platform (it would look for a specific DOM element and dispatch an artificial tap on it, which is brittle and doesn't work on all browsers).

Fixes https://github.com/flutter/flutter/issues/65192

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by ditman — APPROVED** [review]
Does this fix https://github.com/flutter/flutter/issues/65192 ?

**Comment by yjbanov** [comment]
> Does this fix #65192 ?

Yep. Thanks for the heads up. Updated the PR description.

**Comment by ditman** [comment]
```
/tmp/flutter sdk/packages/flutter_driver/lib/src/driver/driver.dart:177: Deprecation notice does not match required pattern.
See: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
```

**Comment by yjbanov** [comment]
> ```
> /tmp/flutter sdk/packages/flutter_driver/lib/src/driver/driver.dart:177: Deprecation notice does not match required pattern.
> See: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
> ```

Done.

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 1]
```diff
@@ -174,8 +174,12 @@ abstract class FlutterDriver {
   async_io.WebDriver get webDriver => throw UnimplementedError();
 
   /// Enables accessibility feature.
+  @Deprecated(
+    'Call setSemantics(true) instead. '
+    'This feature was deprecated after 2.3.0-12.1.pre.'
```
Comment (ditman) [inline packages/flutter_driver/lib/src/driver/driver.dart:179]:
```suggestion
    'This feature was deprecated after v2.3.0-12.1.pre.'
```

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by yjbanov** [comment]
Google testing is broken due to unrelated PR. All failing tests are mobile tests. Shouldn't be affected by this PR. Landing.

**Pull request merged**